.github/workflows/generic-dev.yml

```diff
@@ -22,184 +22,55 @@ jobs:
 # The master branch exclusive jobs will be in a separate
 # workflow file (the osx tests and meson build that is)
 
-  benchmarking:
-    runs-on: ubuntu-latest
+  meson-msvc:
+    name: 'Meson on Windows MSVC'
+    runs-on: windows-latest
     steps:
     - uses: actions/checkout@v2
-    - name: make benchmarking
-      run: make benchmarking
 
-  test:
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v2
-    - name: make test
-      run: make test
+    # Some replacements:
+    # * https://renenyffenegger.ch/notes/Windows/development/Visual-Studio/environment-variables/index
+    # * https://github.com/microsoft/vswhere/wiki/Start-Developer-Command-Prompt
+    - uses: ilammy/msvc-dev-cmd@v1
 
-  gcc-6-7-libzstd:
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v2
-    - name: gcc-6 + gcc-7 + libzstdmt compilation
+    - name: install meson
       run: |
-        make gcc6install gcc7install
-        CC=gcc-6 CFLAGS=-Werror make -j all
-        make clean
-        CC=gcc-7 CFLAGS=-Werror make -j all
-        make clean
-        LDFLAGS=-Wl,--no-undefined make -C lib libzstd-mt
-        make -C tests zbufftest-dll
-
-  gcc-8-asan-ubsan-testzstd:
-    runs-on: ubuntu-16.04 # fails on 18.04
-    steps:
-    - uses: actions/checkout@v2
-    - name: gcc-8 + ASan + UBSan + Test Zstd
+        pip install -U meson
+
+    - run: |
+        git clone https://github.com/lzutao/lz4/ third-party\lz4
+        cd third-party\lz4
```

Comment:
I think lz4 is in vcpkg, like zlib below. Will be great to use the same approach for the dependencies